dev/domain-controller/new-admin-vm/Install-CAM.ps1

```diff
@@ -1320,7 +1330,159 @@ brokerLocale=en_US
+						[System.Environment]::SetEnvironmentVariable("CAM_URI", $camSaasBaseUri, "Machine")
+						Write-Host "Cloud Access Manager PoC has been registered succesfully"
+
+						# Get a Sign-in token
+						$signInResult = ""
+						try {
+							$signInResult = Invoke-RestMethod -Method Post -Uri ($camSaasBaseUri + "/api/v1/auth/signin") -Body $userRequest
+						} catch {
+							$signInResult = ConvertFrom-Json $_.ErrorDetails.Message
+						}
+						Write-Verbose ((ConvertTo-Json $signInResult) -replace "\.*token.*", 'Token": "Sanitized"')
+						# Check if signIn succeded
+						if ($signInResult.code -ne 200) {
+							throw ("Signing in failed. Result was: " + (ConvertTo-Json $signInResult))
+						}
+						$tokenHeader = @{
+							authorization=$signInResult.data.token
+						}
+						Write-Host "Cloud Access Manager sign in succeeded"
+
+						$rcCred = $using:registrationCodeAsCred
+						$registrationCode = $rcCred.Password
+
+						# Register Deployment
+						$deploymentRequest = @{
+							resourceGroup = $using:RGName
+							subscriptionId = $subscription
+							registrationCode = $registrationCode
+						}
+						$registerDeploymentResult = ""
+						try {
+							$registerDeploymentResult = Invoke-RestMethod -Method Post -Uri ($camSaasBaseUri + "/api/v1/deployments") -Body $deploymentRequest -Headers $tokenHeader
+						} catch {
+							$registerDeploymentResult = ConvertFrom-Json $_.ErrorDetails.Message
+						}
+						Write-Verbose ((ConvertTo-Json $registerDeploymentResult) -replace "\.*registrationCode.*", 'registrationCode":"Sanitized"')
+						# Check if registration succeeded
+						if( !( ($registerDeploymentResult.code -eq 201) -or ($registerDeploymentResult.data.reason.ToLower().Contains("already exist")) ) ) {
+							throw ("Registering Deployment failed. Result was: " + (ConvertTo-Json $registerDeploymentResult))
+						}
+						$deploymentId = ""
+						# Get the deploymentId
+						if( ($registerDeploymentResult.code -eq 409) -and ($registerDeploymentResult.data.reason.ToLower().Contains("already exist")) ) {
+							# Deplyoment is already registered so the deplymentId needs to be retrieved
+							$registeredDeployment = ""
+							try {
+								$registeredDeployment = Invoke-RestMethod -Method Get -Uri ($camSaasBaseUri + "/api/v1/deployments") -Body $deploymentRequest -Headers $tokenHeader
+								$deploymentId = $registeredDeployment.data.deploymentId
+							} catch {
+								$registeredDeployment = ConvertFrom-Json $_.ErrorDetails.Message
+								throw ("Getting Deployment ID failed. Result was: " + (ConvertTo-Json $registeredDeployment))
+							}
+						} else {
+							$deploymentId = $registerDeploymentResult.data.deploymentId
+						}
+						[System.Environment]::SetEnvironmentVariable("CAM_DEPLOYMENTID", $deploymentId, "Machine")
+						Write-Host "Deployment has been registered succesfully with Cloud Access Manager"
+
+						# Register Agent Machine
```

Comment:
@fwang-teradici You're right - i think we're accepting it though because we need to provide the password to the java applications somehow